ecommerce/enterprise_coupons/views.py

```
@@ -0,0 +1,54 @@
+"""
+Views for enterprise coupons application.
+"""
+
+import logging
+
+from django.http import HttpResponse
+from django.shortcuts import redirect
+from django.views import View
+
+from ecommerce.enterprise_coupons.backends.base import get_enterprise_catalog_uuid_from_coupon
+from ecommerce.enterprise_coupons.exceptions import CatalogueNotFoundError, CouponError, CouponNotFoundError
+from ecommerce.enterprise_coupons.utils import get_build_mfe_base_url
+
+logger = logging.getLogger(__name__)
+
+
+class CouponRedirectView(View):
+    """
+    View to redirect users to the MFE catalog page based on a coupon code.
+
+    This view extracts the catalog UUID from the provided coupon code and redirects
+    the user to the appropriate MFE catalog page with the catalog UUID as a parameter.
+
+    Returns:
+        - 302: Redirect to MFE catalog page on success
+        - 404: Empty response for coupon or catalogue not found
+        - 400: Empty response for invalid coupon
+        - 500: Empty response for configuration issues
+
+    Note:
+        All errors are logged internally with detailed information for debugging,
+        but NO messages are returned to clients to avoid exposing implementation
+        details or providing information to potential attackers.
+    """
+    def get(self, request, coupon_code):
```

What do you think about this: https://github.com/Pearson-Advance/ecommerce/pull/46#discussion_r2611353235? @daviidco I think we need to return a default response to avoid what I mentioned there.

---

Good point! I’ve added a catch-all exception handler and a new setting. Let me know if this approach works for you @Squirrel18